packages/@sanity/core/src/actions/start/startAction.js

```diff
@@ -138,8 +208,71 @@ function printWarnings(output, warnings) {
   output.print(chalk.yellow('Compiled with warnings.'))
   output.print()
 
-  warnings.map(message => `Warning in ${formatMessage(message)}`).forEach(message => {
-    output.print(message)
-    output.print()
+  warnings
+    .map(message => `Warning in ${formatMessage(message)}`)
+    .forEach(message => {
+      output.print(message)
+      output.print()
+    })
+}
+
+async function getOrCreateProject(context) {
+  const {prompt, apiClient} = context
+
+  let projects
+  try {
+    projects = await apiClient({requireProject: false}).projects.list()
+  } catch (err) {
+    throw new Error(`Failed to communicate with the Sanity API:\n${err.message}`)
+  }
+
+  if (projects.length === 0) {
+    debug('No projects found for user, prompting for name')
+    const projectName = await prompt.single({message: 'Project name'})
+    return createProject(apiClient, {displayName: projectName})
+  }
+
+  debug(`User has ${projects.length} project(s) already, showing list of choices`)
+
+  const projectChoices = projects.map(project => ({
+    value: project.id,
+    name: `${project.displayName} [${project.id}]`
+  }))
+
+  const selected = await prompt.single({
+    message: 'Select project to use',
+    type: 'list',
+    choices: [{value: 'new', name: 'Create new project'}, new prompt.Separator(), ...projectChoices]
+  })
+
+  if (selected === 'new') {
+    debug('User wants to create a new project, prompting for name')
+    return createProject(apiClient, {
+      displayName: await prompt.single({
+        message: 'Informal name for your project'
+      })
+    })
+  }
+
+  debug(`Returning selected project (${selected})`)
+  return {
+    projectId: selected,
+    displayName: projects.find(proj => proj.id === selected).displayName
+  }
+}
+
+function createProject(apiClient, options) {
```

Comment:
Any reason this isn't async like the rest of the functions in here?

Reply:
Just reusing existing work. Copy + paste etc.